ssz/sedes/serializable.py

```diff
@@ -0,0 +1,419 @@
+import abc
+import collections
+import copy
+import re
+
+from eth_utils import (
+    to_dict,
+    to_set,
+    to_tuple,
+)
+
+from ssz.constants import (
+    CONTAINER_PREFIX_LENGTH,
+)
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+
+
+class MetaBase:
+    fields = None
+    field_names = None
+    field_attrs = None
+
+
+def _get_duplicates(values):
+    counts = collections.Counter(values)
+    return tuple(
+        item
+        for item, num in counts.items()
+        if num > 1
+    )
+
+
+def validate_args_and_kwargs(args, kwargs, arg_names, allow_missing=False):
+    duplicate_arg_names = _get_duplicates(arg_names)
+    if duplicate_arg_names:
+        raise TypeError("Duplicate argument names: {0}".format(sorted(duplicate_arg_names)))
+
+    needed_kwargs = arg_names[len(args):]
+    used_kwargs = set(arg_names[:len(args)])
+
+    duplicate_kwargs = used_kwargs.intersection(kwargs.keys())
+    if duplicate_kwargs:
+        raise TypeError("Duplicate kwargs: {0}".format(sorted(duplicate_kwargs)))
+
+    unknown_kwargs = set(kwargs.keys()).difference(arg_names)
+    if unknown_kwargs:
+        raise TypeError("Unknown kwargs: {0}".format(sorted(unknown_kwargs)))
+
+    missing_kwargs = set(needed_kwargs).difference(kwargs.keys())
+    if not allow_missing and missing_kwargs:
+        raise TypeError("Missing kwargs: {0}".format(sorted(missing_kwargs)))
+
+
+@to_tuple
+def merge_kwargs_to_args(args, kwargs, arg_names, allow_missing=False):
```

Comment:
I'm a bit surprised that there's both `merge_kwrags_to_args` and `merge_args_to_kwargs`, I suspect we can get rid of one. Working with `kwargs` feels safer, so I'd vote to keep `merge_args_to_kwargs`.

Reply:
I think `merge_kwargs_to_args` function is needed because it takes in both `args` and `kwargs` and returns the `args` form of them combined. Like some of the fields may be given as `args`, whereas the rest maybe given as `kwargs`, but this function takes all of them and returns the `args` form of them.